### DISCUSSION TITLE
com.sun.javafx.css.converters.EnumConverter missing in javafx18

### DISCUSSION THREAD

**Discussion by sangiorgi-inaf** [GSI-CS-CO/chart-fx#591] [Q&A] [answered]
Hi, I've an old project which use chart-fx 8.1.5.
Now I had to update the runtime library from java 1.8 to java 17 and external JavaFx 18.
Code related to javafx works, but I've an exception of java.lang.ClassNotFoundException: com.sun.javafx.css.converters.EnumConverter at this line:
final DefaultNumericAxis xAxis = new DefaultNumericAxis("");

Is there a way to run the code with chart-fx 8.1.5 + jdk17 + javafx 18 using some arguments in run configuration or something similar or do I have necessary to change chart-fx library?

thanks in advance

**Comment by wirew0rm** [comment, marked as answer]
This is definitely not supported and I would recommend to upgrade to the latest chart-fx version, which also contains lots of improvements in other areas.
If you cannot change the code at all, you can just switch to 11.1.5 which is exactly the same as 8.1.5 minus the [backport commit](https://github.com/fair-acc/chart-fx/commit/fd22cafef14319e484623eeb1203b50928f55658) for java8.
